storage/storage_model.go

```diff
@@ -68,6 +85,28 @@ func (info *LogChunkInfo) fromKey(path string) error {
 	return nil
 }
 
+func (info *LogChunkInfo) fromLogChunk(buildID string, testID string, logChunk model.LogChunk) error {
+	if len(logChunk) == 0 {
+		return errors.New("Log chunk must contain at least one line")
+	}
+	minTime := TimeRangeMax
+	maxTime := TimeRangeMin
+	for _, logLine := range logChunk {
```

Comment:
Should we ensure that log lines are in sorted ts order, at least within a chunk?

Reply:
So the current implementation just assumes that it is, without ensuring that. I'm not inclined to do additional sorting at least in this change since it would be a significant functional change if people are sending us stuff out of order.

Reply:
If we are assuming then do we need to find the min and the max? Can't we just grab the first and last timestamp from the chunk?